### PR TITLE
Destination Iceberg: Fix tests to run in airbyte-ci

### PR DESCRIPTION
**Pull request by gisripa**
## What
Fixing tests, reconfiguring docker ports to run in airbyte-ci environment.

## Review guide
* Removed docker-compose dependency for bootstrapping rest catalog containers. Converted them to individual docker containers
* Disabled Hive related tests and renamed its dependency to `MinioContainerToDelete`.
* New `MinioContainer` for reusing the dependency in other catalogs.
* Quick fixes in config for `host-port` to use container network vs host network depending on connector run calls vs test runner calls. 

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by gisripa** [comment]
* **#45206** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/45206?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gisripa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-iceberg/src/test-integration/java/io/airbyte/integrations/destination/iceberg/hadoop/BaseIcebergHadoopCatalogS3IntegrationTest.java [hunk 1]
```diff
@@ -103,9 +96,21 @@ protected List<JsonNode> retrieveRecords(final TestDestinationEnv testEnv,
...
+    return IcebergIntegrationTestUtil.retrieveRecords(getConfigForTestRunner(), namespace, streamName);
   }
 
   abstract DataFileFormat fileFormat();
 
+  @Nullable
+  @Override
+  protected String getDefaultSchema(@NotNull JsonNode config) throws Exception {
+    // TODO: This was NPE'ing without this return value because of Kotlin's non-null in base,
+    // but whats the actual value to pass instead of empty ?
```
Comment (stephane-airbyte) [inline airbyte-integrations/connectors/destination-iceberg/src/test-integration/java/io/airbyte/integrations/destination/iceberg/hadoop/BaseIcebergHadoopCatalogS3IntegrationTest.java:108]: make the function abstract?

**Review by stephane-airbyte — APPROVED** [review]

File: airbyte-integrations/connectors/destination-iceberg/src/test-integration/java/io/airbyte/integrations/destination/iceberg/hadoop/BaseIcebergHadoopCatalogS3IntegrationTest.java (hunk 1 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-iceberg/src/test-integration/java/io/airbyte/integrations/destination/iceberg/hadoop/BaseIcebergHadoopCatalogS3IntegrationTest.java:108]: where are we inheriting this as nonnullable from? DAT.kt says it returns `String?` on master https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/DestinationAcceptanceTest.kt#L261

**Review by edgao — APPROVED** [review]
rubberstamping

File: airbyte-integrations/connectors/destination-iceberg/src/test-integration/java/io/airbyte/integrations/destination/iceberg/hadoop/BaseIcebergHadoopCatalogS3IntegrationTest.java (hunk 1 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-iceberg/src/test-integration/java/io/airbyte/integrations/destination/iceberg/hadoop/BaseIcebergHadoopCatalogS3IntegrationTest.java:108]: Bad comment, actually was doing `!!` in actual tests for defaultSchema which was set as null in base class.

**Pull request merged**